lib/tpm2_attr_util.c

```diff
@@ -242,39 +239,151 @@ static bool nt(TPMA_NV *nv, char *arg) {
-    dispatch_arg_add(nt, 3),             //  5
-    dispatch_arg_add(nt, 2),             //  6
-    dispatch_arg_add(nt, 1),             //  7
-    dispatch_reserved(8),                //  8
-    dispatch_reserved(9),                //  9
-    dispatch_no_arg_add(policydelete),   // 10
-    dispatch_no_arg_add(writelocked),    // 11
-    dispatch_no_arg_add(writeall),       // 12
-    dispatch_no_arg_add(writedefine),    // 13
-    dispatch_no_arg_add(write_stclear),  // 14
-    dispatch_no_arg_add(globallock),     // 15
-    dispatch_no_arg_add(ppread),         // 16
-    dispatch_no_arg_add(ownerread),      // 17
-    dispatch_no_arg_add(authread),       // 18
-    dispatch_no_arg_add(policyread),     // 19
-    dispatch_reserved(20),               // 20
-    dispatch_reserved(21),               // 21
-    dispatch_reserved(22),               // 22
-    dispatch_reserved(23),               // 23
-    dispatch_reserved(24),               // 24
-    dispatch_no_arg_add(no_da),          // 25
-    dispatch_no_arg_add(orderly),        // 26
-    dispatch_no_arg_add(clear_stclear),  // 27
-    dispatch_no_arg_add(readlocked),     // 28
-    dispatch_no_arg_add(written),        // 29
-    dispatch_no_arg_add(platformcreate), // 30
-    dispatch_no_arg_add(read_stclear),   // 31
+static dispatch_table nv_attr_table[] = { // Bit Index
+    dispatch_no_arg_add(ppwrite),         //  0
+    dispatch_no_arg_add(ownerwrite),      //  1
+    dispatch_no_arg_add(authwrite),       //  2
+    dispatch_no_arg_add(policywrite),     //  3
+    dispatch_arg_add(nt, 4),              //  4
+    dispatch_arg_add(nt, 3),              //  5
+    dispatch_arg_add(nt, 2),              //  6
+    dispatch_arg_add(nt, 1),              //  7
+    dispatch_reserved(8),                 //  8
+    dispatch_reserved(9),                 //  9
+    dispatch_no_arg_add(policydelete),    // 10
+    dispatch_no_arg_add(writelocked),     // 11
+    dispatch_no_arg_add(writeall),        // 12
+    dispatch_no_arg_add(writedefine),     // 13
+    dispatch_no_arg_add(write_stclear),   // 14
+    dispatch_no_arg_add(globallock),      // 15
+    dispatch_no_arg_add(ppread),          // 16
+    dispatch_no_arg_add(ownerread),       // 17
+    dispatch_no_arg_add(authread),        // 18
+    dispatch_no_arg_add(policyread),      // 19
+    dispatch_reserved(20),                // 20
+    dispatch_reserved(21),                // 21
+    dispatch_reserved(22),                // 22
+    dispatch_reserved(23),                // 23
+    dispatch_reserved(24),                // 24
+    dispatch_no_arg_add(no_da),           // 25
+    dispatch_no_arg_add(orderly),         // 26
+    dispatch_no_arg_add(clear_stclear),   // 27
+    dispatch_no_arg_add(readlocked),      // 28
+    dispatch_no_arg_add(written),         // 29
+    dispatch_no_arg_add(platformcreate),  // 30
+    dispatch_no_arg_add(read_stclear),    // 31
+};
+
+static bool fixedtpm(TPMA_OBJECT *obj, char *arg) {
+
+    (void) arg;
```

Comment:
Wasn't the convention in the project to use the UNUSED() macro for this?

Now, it seems we are using this macro in a lot of places so I wonder if we shouldn't just add a `-Wno-unused-parameter` to the `EXTRA_CFLAGS` in `configure.ac` or something along those lines since we are silent the warning anyways by casting to void.

Reply:
Yes that is correct. I'm not a big fan of adding compiler flags to undo warnings. I'd rather the code be explicit about an unused variable.